src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -331,78 +335,55 @@ public SnowflakeStreamingIngestChannelInternal<?> openChannel(OpenChannelRequest
-              .setChannelSequencer(response.getClientSequencer())
-              .setOwningClient(this)
-              .setEncryptionKey(response.getEncryptionKey())
-              .setEncryptionKeyId(response.getEncryptionKeyId())
-              .setOnErrorOption(request.getOnErrorOption())
-              .setDefaultTimezone(request.getDefaultTimezone())
-              .setOffsetTokenVerificationFunction(request.getOffsetTokenVerificationFunction())
-              .build();
-
-      // Setup the row buffer schema
-      channel.setupSchema(response.getTableColumns());
+    OpenChannelRequestInternal openChannelRequest =
+        new OpenChannelRequestInternal(
+            this.storageManager.getClientPrefix() + "_" + counter.getAndIncrement(),
+            this.role,
+            request,
+            Constants.WriteMode.CLOUD_STORAGE,
+            isIcebergMode);
+    OpenChannelResponse response = snowflakeServiceClient.openChannel(openChannelRequest);
 
-      // Add channel to the channel cache
-      this.channelCache.addChannel(channel);
+    logger.logInfo(
+        "Open channel request succeeded, channel={}, table={}, clientSequencer={},"
+            + " rowSequencer={}, client={}",
+        request.getChannelName(),
+        request.getFullyQualifiedTableName(),
+        response.getClientSequencer(),
+        response.getRowSequencer(),
+        getName());
 
-      return channel;
-    } catch (IOException | IngestResponseException e) {
-      throw new SFException(e, ErrorCode.OPEN_CHANNEL_FAILURE, e.getMessage());
-    }
+    // Channel is now registered, add it to the in-memory channel pool
+    SnowflakeStreamingIngestChannelInternal<T> channel =
+        SnowflakeStreamingIngestChannelFactory.<T>builder(response.getChannelName())
+            .setDBName(response.getDBName())
+            .setSchemaName(response.getSchemaName())
+            .setTableName(response.getTableName())
+            .setOffsetToken(response.getOffsetToken())
+            .setRowSequencer(response.getRowSequencer())
+            .setChannelSequencer(response.getClientSequencer())
+            .setOwningClient(this)
+            .setEncryptionKey(response.getEncryptionKey())
+            .setEncryptionKeyId(response.getEncryptionKeyId())
+            .setOnErrorOption(request.getOnErrorOption())
+            .setDefaultTimezone(request.getDefaultTimezone())
+            .setOffsetTokenVerificationFunction(request.getOffsetTokenVerificationFunction())
+            .build();
+
+    // Setup the row buffer schema
+    channel.setupSchema(response.getTableColumns());
+
+    // Add channel to the channel cache
+    this.channelCache.addChannel(channel);
+
+    // Add storage to the storage manager, only for external volume
+    this.storageManager.addStorage(
+        response.getDBName(),
+        response.getSchemaName(),
+        response.getTableName(),
+        response.getStageLocation());
+
+    return channel;
   }
```

Comment:
i'd bring back this catch block as it was covering more logic (open_channel API call + local bookkeeping) than what you have now (just the API call).

Reply:
Reverted.